pkg/object/rawconfigtrafficcontroller/rawconfigtrafficcontroller.go

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package rawconfigtrafficcontroller
+
+import (
+	"fmt"
+
+	"github.com/megaease/easegress/pkg/logger"
+	"github.com/megaease/easegress/pkg/object/httppipeline"
+	"github.com/megaease/easegress/pkg/object/httpserver"
+	"github.com/megaease/easegress/pkg/object/trafficcontroller"
+	"github.com/megaease/easegress/pkg/supervisor"
+)
+
+const (
+	// Category is the category of RawConfigTrafficController.
+	Category = supervisor.CategorySystemController
+
+	// Kind is the kind of RawConfigTrafficController.
+	Kind = "RawConfigTrafficController"
+
+	DefaultNamespace = "default"
+)
+
+type (
+	// RawConfigTrafficController is a system controller to manage
+	// TrafficGate, Pipeline and their relationship.
+	RawConfigTrafficController struct {
```

Comment:
Is that the difference between`RawConfigTrafficController` and `TrafficController` is only the default-namespace and mulitple-namespace? If so, proposal to remove this `RawConfigTrafficController` and let `TrafficController` handle the default namespace as well.

Reply:
https://github.com/megaease/easegress/pull/55/files?short_path=f277e23#diff-f277e23e848049e2b25d0b86693cdcecae6c031e2d70271a4715f2e4d26f5f21

`RawConfigTrafficController` is a must-have controller to decouple `TrafficController` with the input source. It will bring more complicity and less extensibility to `TrafficController`.